### PR TITLE
Task/refactor action callback

### LINKED ISSUE #2634: Rename Action to Callback and Callback to JSCallback
At PyData London, @bryevdv and I got into this weird disagreement about a new callback. After a while I realized that I was just confused by the current naming structure.

Right now in https://github.com/bokeh/bokeh/blob/master/bokeh/models/actions.py

we have

```
- Action(PlotObject)
-- Callback(Action)
-- OpenURL(Action)
```

In most of the codebase where we have callbacks, we have

`callback = Instance(Callback, ....)`

but on TapTool we have 

`action = Instance(Action,...)`

with the example centered around OpenURL. Of course, action could actually be a Callback, as Callback is also a subclass of Action.

This inconsistency is just a historical accident as OpenURL came before Callback.

I would like to propose the following, backwards incompatible, changes.

1 - rename `Action` to `Callback` and `Callback` to `CustomJS`
```
- Callback(PlotObject)
-- OpenURL(Callback)
-- CustomJS(Callback)
```

2 - replace TapTool `action` kwarg to be `callback`.

All models with callbacks have the property as previously:

`callback = Instance(Callback, ....)`

(of course, this is now referring to the higher level class).

Given that we have been talking about new "`JupyterCallback`"s and "`ServerCallback`"s I imagine that these would also subclass Callback and be acceptable inputs to a model that accepts a callback.  So for example you could imagine all of these as possible future definitions of a TapTool

```python
customjs_tap = TapTool(callback=CustomJS(args={}, code="alert('I was tapped');"))
openurl_tap = TapTool(callback=OpenURL(url="http://openontap.com"))
jupyter_tap = TapTool(callback=Jupyter(args={}, method=lambda: print('i was tapped')))
```

Given that `Callback` and even `Action` are pretty new, I think this is the best time to do it before we start adding a lot more canned "callbacks."

### PR DESCRIPTION
**Pull request by birdsarah**
Fixes #2634

With the deprecation functions in place, I believe this is non-breaking. 

There are tests for the deprecation and I have confirmed manually that, for example, you can still use `from bokeh.models.actions import Callback` and everything works as expected.

### CODE REVIEW

**Comment by birdsarah** [comment]
Note this will break other language bindings, as Callback has changed on the JS side without deprecation.

File: bokeh/tests/test_actions_deprecation.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import unittest
+import warnings
+
+from bokeh.models import Action as modelsAction
+from bokeh.models.actions import Callback, OpenURL, Action
+from bokeh.models import callbacks
+
+
+class ActionsDeprecationTests(unittest.TestCase):
```
Comment (bryevdv) [inline bokeh/tests/test_actions_deprecation.py]: Until we actually get to a point where we offer selectable compatibility modes (`bokeh.set_version("0.11.2")`), and have to e.g., verify bug/behaviour compatibility in different selected versions, I think tests like this are probably more work than useful.

**Comment by birdsarah** [comment]
@bryevdv - would you like me to remove the tests?

@pzwang - given your comment on #2715 - wondering if you have comments/thoughts about this renaming.

**Comment by damianavila** [comment]
This LGTM... 
I would probably remove the tests (as per reasons exposed by @bryevdv), but if you think is better to keep it, I am OK with that too... I will wait a little bit more for @pzwang, but I plan to merge this by EOD/early tomorrow... unless someone else wants more time.

**Pull request merged**